### PR TITLE
Add metric methods to the Group model

### PR DESCRIPTION
**Pull request by adam-thomas**
This is for proper display of the 'last comment', '# participants' and '# discussions' metrics in the list of groups.

@incuna/backend review please? :)

### CODE REVIEW

File: groups/models.py [hunk 1]
```diff
@@ -49,6 +49,18 @@ def unsubscribe(self, user):
...
         return self.watchers.filter(id=user.pk).exists()
 
+    def get_all_comments(self):
+        return self.discussions.comments()
+
+    def get_latest_comment(self):
+        return self.get_all_comments().latest('date_created')
+
+    def get_total_commenters(self):
+        return self.get_all_comments().users().count()
```
Comment (meshy) [inline groups/models.py]: Does this need a distinct?

**Comment by meshy** [comment]
LGTM barring my comment

File: groups/models.py (hunk 1 above)
Comment (adam-thomas) [inline groups/models.py]: Oh yes, thank you!

Comment (adam-thomas) [inline groups/models.py]: Weirdly, I already had a test that claimed it already had protection from duplicates. I don't know if that test is broken, or if `count()` comes with a `distinct()` built-in, or what.

Comment (adam-thomas) [inline groups/models.py]: False alarm - the `users()` method has a `distinct()` in it.

**Comment by meshy** [comment]
:+1:

**Comment by adam-thomas** [comment]
Thanks :)

**Comment by Ian-Foote** [comment]
A minor point, but I dislike the `get_` prefix to these methods. It's just extra verbosity.

**Comment by adam-thomas** [comment]
Fair.  I tend to use it to denote methods that have to do some work (queryset retrievals in this case) to retrieve the data.

**Comment by Ian-Foote** [comment]
I think it's implicit that methods on a Manager will do work.

**Comment by Ian-Foote** [comment]
I think that's implicit in the fact that these are methods.

**Comment by adam-thomas** [comment]
I suppose, although that's not so visible in a template.

**Comment by adam-thomas** [comment]
(Which is what these are intended for.)

**Pull request merged**